lib/datadog/appsec/contrib/rack/gateway/watcher.rb

```diff
@@ -152,8 +153,47 @@ def watch_request_body(gateway = Instrumentation.gateway)
                 end
               end
 
+              def watch_user_id(gateway = Instrumentation.gateway)
```

Comment:
I don't think this belongs in `contrib/rack` as it has nothing Rack specific about it.